src/crimson/osd/pg_backend.cc

```diff
@@ -991,6 +991,12 @@ PGBackend::remove(ObjectState& os, ceph::os::Transaction& txn,
 	     ghobject_t{os.oi.soid, ghobject_t::NO_GEN, shard});
   delta_stats.num_bytes -= os.oi.size;
 
+  // todo: update watchers
+  if (os.oi.is_whiteout()) {
```

Comment:
Can you explain why the previous check location was wrong?
It differs from `PrimaryLogPG::_delete_oid`

Reply:
Oh, I missed the ```if (whiteout && os.oi.is_whiteout()) {``` above -- this change doesn't actually have any effect.  I'll remove it.